subsys/usb/device/class/audio/audio.c

```diff
@@ -541,6 +541,73 @@ static int handle_fu_mute_req(struct usb_audio_dev_data *audio_dev_data,
 	return -EINVAL;
 }
 
+/**
+ * @brief Handler for feature unit volume control requests.
+ *
+ * This function handles feature unit volume control request.
+ *
+ * @param audio_dev_data USB audio device data.
+ * @param setup          Information about the executed request.
+ * @param len		 Size of the buffer.
+ * @param data		 Buffer containing the request result.
+ * @param evt		 Feature Unit Event info.
+ * @param device	 Device part that has been addressed. Applicable for
+ *			 bidirectional device.
+ *
+ * @return 0 if successful, negative errno otherwise.
+ */
+
+static int handle_fu_volume_req(struct usb_audio_dev_data *audio_dev_data,
+			      struct usb_setup_packet *setup,
+			      int32_t *len, uint8_t **data,
+			      struct usb_audio_fu_evt *evt,
+			      uint8_t device)
+{
+	uint8_t ch = (setup->wValue) & 0xFF;
+	uint8_t ch_cnt = audio_dev_data->ch_cnt[device];
+	uint8_t *controls = audio_dev_data->controls[device];
+	uint8_t *control_val = &controls[POS(VOLUME, ch, ch_cnt)];
```

Comment:
Where is VOLUME defined?

Reply:
In `zephyr/subsys/usb/device/class/audio/usb_audio_internal.h` 

You can find these codes.
```

#define POS_MUTE		   0
#define POS_VOLUME		   (POS_MUTE + FEATURE_MUTE_SIZE)
...

#define POS(prop, ch_idx, ch_cnt) (ch_cnt * POS_##prop + \
				  (ch_idx * FEATURE_##prop##_SIZE))
```
It prepends "POS" to prop, so the define is POS_VOLUME